src/client/app/shared/asy-http.service.ts

```diff
@@ -31,124 +31,101 @@ export class AsyHttp {
 
 	constructor(
 		private userStateService: UserStateService,
-		private _http: Http,
+		private _http: HttpClient,
 		private router: Router,
-		private location: Location) {}
+		private location: Location
+	) {}
 
 	static defaultErrFn(_err: any) { }
 
 	get(opts: HttpOptions) {
-		let headers = new Headers({
+		let headers = new HttpHeaders({
 			'Interface-URL': this.getUriEncodedPath()
 		});
 
-		let observable = this._http.get(opts.url, {search: opts.urlParams, headers: headers} )
-			.map((res) => this.hasContent(res) ? res.json() : null)
+		let observable = this._http.get(opts.url, { params: opts.urlParams, headers: headers } )
 			.share()
-			.catch((error: any, caught: Observable<any>) => {
-				return this.handleErrorResponse(error, caught);
+			.catch((error: HttpErrorResponse) => this.handleErrorResponse(error));
 
-			});
-
-		observable
-			.subscribe(
-				(data) => opts.dataFn(data),
-				(err) => opts.errFn(err),
-				() => opts.completeFn()
-			);
+		observable.subscribe(
+			(data: any) => opts.dataFn(data),
+			(err: any) => opts.errFn(err),
+			() => opts.completeFn()
+		);
 
 		return observable;
 	}
 
 	post(opts: HttpOptions) {
-		let headers = new Headers({
+		let headers = new HttpHeaders({
 			'Content-Type': 'application/json',
 			'Interface-URL': this.getUriEncodedPath()
 		});
 
-		let observable = this._http.post(opts.url, JSON.stringify(opts.data), {
-			headers: headers
-		})
-			.map((res) => this.hasContent(res) ? res.json() : null)
+		let observable = this._http.post(opts.url, JSON.stringify(opts.data), { headers: headers })
 			.share()
-			.catch((error: any, caught: Observable<any>) => {
-				return this.handleErrorResponse(error, caught);
-			});
-
-		observable
-			.subscribe(
-				(data) => opts.dataFn(data),
-				(err) => opts.errFn(err),
-				() => opts.completeFn()
-			);
+			.catch((error: HttpErrorResponse) => this.handleErrorResponse(error));
+
+		observable.subscribe(
```

Comment:
Do we need to unsubscribe from these somehow, or is that handled implicitly (somehow)?

Reply:
I don't think we're unsubscribing from them implicitly or explicitly, unfortunately. This service was written when our knowledge of angular and observables was much newer. I think in the future, we might want to take advantage of the HttpInterceptor interface that was introduced in Angular4 instead of using this.